src/guards/member-major.guard.spec.ts

```diff
@@ -0,0 +1,139 @@
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should be defined', () => {
+    expect(memberMajorGuard).toBeDefined();
+  });
+
+  describe('canActivate', () => {
+    let allowMemberMajors: MemberMajors;
+    let member: MemberEntity;
+
+    beforeEach(() => {
+      member = new MemberEntity();
+    });
+
+    describe('major 미선택 member 를 허용 ', () => {
+      beforeEach(() => {
+        allowMemberMajors = [];
+        mockReflector.get.mockReturnValue(allowMemberMajors);
+      });
+
+      it('성공 케이스', () => {
+        member.majorId = null;
+        mockRequest.user = member;
+
+        expect(memberMajorGuard.canActivate(mockContext)).toBeTruthy();
+      });
+
+      it('실패 케이스', () => {
+        member.majorId = MajorId.Design;
+        mockRequest.user = member;
+
+        expect(() => {
+          memberMajorGuard.canActivate(mockContext);
+        }).toThrowError('major 미선택 유저만 접근 가능합니다.');
+      });
+    });
+
+    describe('major develop member 를 허용 ', () => {
+      beforeEach(() => {
+        allowMemberMajors = [MajorId.Development];
+        mockReflector.get.mockReturnValue(allowMemberMajors);
+      });
+
+      it('성공 케이스', () => {
+        member.majorId = MajorId.Development;
+        mockRequest.user = member;
+
+        expect(memberMajorGuard.canActivate(mockContext)).toBeTruthy();
+      });
+
+      it('실패 케이스', () => {
+        member.majorId = MajorId.Design;
+        mockRequest.user = member;
+
+        expect(() => {
+          memberMajorGuard.canActivate(mockContext);
+        }).toThrowError('Development major 유저만 접근 가능합니다.');
+      });
+    });
+
+    describe('major design member 를 허용 ', () => {
+      beforeEach(() => {
+        allowMemberMajors = [MajorId.Design];
+        mockReflector.get.mockReturnValue(allowMemberMajors);
+      });
+
+      it('성공 케이스', () => {
+        member.majorId = MajorId.Design;
+        mockRequest.user = member;
+
+        expect(memberMajorGuard.canActivate(mockContext)).toBeTruthy();
+      });
+
+      it('실패 케이스', () => {
+        member.majorId = MajorId.Development;
+        mockRequest.user = member;
+
+        expect(() => {
+          memberMajorGuard.canActivate(mockContext);
+        }).toThrowError('Design major 유저만 접근 가능합니다.');
+      });
+    });
+
+    describe('major design, develop member 를 허용 ', () => {
+      beforeEach(() => {
+        allowMemberMajors = [MajorId.Development, MajorId.Design];
+        mockReflector.get.mockReturnValue(allowMemberMajors);
+      });
+
+      it('성공 케이스', () => {
+        member.majorId = MajorId.Development;
+        mockRequest.user = member;
+
+        expect(memberMajorGuard.canActivate(mockContext)).toBeTruthy();
+      });
+
+      it('실패 케이스', () => {
+        member.majorId = null;
+        mockRequest.user = member;
+
+        expect(() => {
+          memberMajorGuard.canActivate(mockContext);
+        }).toThrowError('Development, Design major 유저만 접근 가능합니다.');
```

Comment:
```suggestion
 describe('특정 major를 선택한 member만 허용 ', () => {
      beforeEach(() => {
        majorId = faker.helpers.arrayElement([
          MajorId.Development,
          MajorId.Design,
        ]);
        allowMemberMajors = [majorId];

        mockReflector.get.mockReturnValue(allowMemberMajors);
      });

      it('성공 케이스', () => {
        member.majorId = majorId;
        mockRequest.user = member;

        expect(memberMajorGuard.canActivate(mockContext)).toBeTruthy();
      });

      it('실패 케이스', () => {
        member.majorId = null;
        mockRequest.user = member;

        expect(() => {
          memberMajorGuard.canActivate(mockContext);
        }).toThrowError(`${MajorId[majorId]} major 유저만 접근 가능합니다.`);
      });
    });
```
  faker.helpers.arrayElement([])를 사용하는 방식은 어떻게 생각하시나요? 
 "faker.heplers.arrayElements"를 이용하면 배열에서 선택하고 싶은 개수도 고를 수있는 것 같네요 it.each와 응용하면 좋을 것 같아요

Reply:
오 faker 에 그런것도 있었네요.
하지만 여기서 사용하게되면 어떤 랜덤한 값이 뽑혔을 때 분기처리를 테스트코드 내에서 해야할거같아서 부적절할거같아요
테스트코드에서는 최대한 if 문 같은 논리적인 코드를 피해야한다고 본 기억이 있어서 혹시 arrayElements 를 사용해서 깔끔하게 짤 수 있는 방법 있으면 공유해주시면 감사하겠습니다.

Reply:
```suggestion
   describe('test', () => {
  it.each([0, 1, 2])('성공케이스 - count: %s', (count) => {
    majorId = faker.helpers.arrayElements(
      [MajorId.Development, MajorId.Design],
      count,
    );
    allowMemberMajors = [majorId];

    mockReflector.get.mockReturnValue(allowMemberMajors);

    member.majorId = majorId;
    mockRequest.user = member;

    expect(memberMajorGuard.canActivate(mockContext)).toBeTruthy();
  });
});
```
약간 요런 느낌을 말씀드린거였슴다 ㅎㅎ